src/engine/tools/engine-suite/src/engine_test/formats/multi_line.py

```diff
@@ -13,19 +13,19 @@ def format_event(self, event):
         return event
 
     def get_events(self, events):
-        lines = 0
-        event = ''
+
         events_formated = []
-        for line in events:
-            event = event + line
-            lines = lines + 1
-            # Only add if the number of lines is full, the rest are discarded
-            if lines == self.config['lines']:
-                events_formated.append(event)
-                event = ''
-                lines = 0
-            else:
-                event = event + ' '
+        maxLines = int(self.config['lines'])
+        # Group events by maxLines
+        for event in events:
+            clean_event = event.strip()
+            # Split event by lines
+            lines = clean_event.splitlines()
+            chunks = []
+            for i in range(0, len(lines), maxLines):
```

Comment:
improve - design: i think that yo could delete the need of manually check if chunck have a specific number lines:

```
def get_events(self, events):
    maxLines = int(self.config['lines'])
    events_formated = []

    for event in events:
        # split event in lines 
        lines = event.strip().splitlines()

        # group lines into chunks of maximum size maxLines
        chunks = [lines[i:i + maxLines] for i in range(0, len(lines), maxLines)]

        # join the lines of each chunck in an only formatted event
        events_formated.extend([' '.join(chunk) for chunk in chunks])

    return events_formated
```

Reply:
done!